auth/auth/auth.py

```diff
@@ -263,6 +272,66 @@ async def callback(request):
     return aiohttp.web.HTTPFound(next_page)
 
 
+@routes.post('/api/v1alpha/user/{user}/create')
+@rest_authenticated_developers_only
+async def create_user(request: web.Request, userdata):  # pylint: disable=unused-argument
+    db: Database = request.app['db']
+    username = request.match_info['user']
+
+    body = await request.json()
+    login_id = body['login_id']
+    is_developer = body['is_developer']
+    is_service_account = body['is_service_account']
+
+    if not username:
+        raise web.HTTPBadRequest(body='username must be a non-zero length string')
+    if not isinstance(login_id, str) or not login_id:
+        raise web.HTTPBadRequest(body='login_id must be a non-zero length string')
+    if not isinstance(is_developer, bool):
+        raise web.HTTPBadRequest(body='invalid value specified for is_developer')
+    if not isinstance(is_service_account, bool):
+        raise web.HTTPBadRequest(body='invalid value specified for is_service_account')
+    if is_developer and is_service_account:
+        raise web.HTTPBadRequest(body='user cannot be both a developer and a service account')
+
+    existing_users = [
+        x
+        async for x in db.select_and_fetchall(
+            "SELECT * FROM users WHERE login_id = %s OR username = %s;", (login_id, username)
+        )
+    ]
+    if len(existing_users) == 1:
+        existing_user = existing_users[0]
+        expected_username = existing_user['username']
+        expected_login_id = existing_user['login_id']
+        if username != expected_username:
+            raise web.HTTPBadRequest(body=f"user '{expected_username}' already exists with login id '{login_id}'")
+        if login_id != expected_login_id:
+            raise web.HTTPBadRequest(
+                body=f"user '{username}' already exists with a different login id '{expected_login_id}'"
+            )
+    elif len(existing_users) == 0:
+        await db.execute_insertone(
+            '''
+        INSERT INTO users (state, username, login_id, is_developer)
```

Comment:
Thanks for putting this together @jigold! It doesn't look like this handles the `is_service_account` case: https://github.com/hail-is/hail/blob/9aa1e6acd7e2e6c443f0c501ae6c97359641f50c/auth/auth/auth.py#L465-L466

Is it worth combining the logic within these two functions? It doesn't look like `post_create_user` explicitly checks for existing users.